### PR TITLE
[SPARK-40663][SQL] Migrate execution errors onto error classes: _LEGACY_ERROR_TEMP_2201-2225

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to migrate 25 execution errors onto temporary error classes with the prefix `_LEGACY_ERROR_TEMP_2201` to `_LEGACY_ERROR_TEMP_2225`.

The error classes are prefixed with `_LEGACY_ERROR_TEMP_` indicates the dev-facing error messages, and won't be exposed to end users.

### Why are the changes needed?

To speed-up the error class migration.

The migration on temporary error classes allow us to analyze the errors, so we can detect the most popular error classes.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

```
$ build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
$ build/sbt "test:testOnly *SQLQuerySuite"
$ build/sbt -Phive-thriftserver "hive-thriftserver/testOnly org.apache.spark.sql.hive.thriftserver.ThriftServerQueryTestSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@itholic Could you rebase it on the recent master, please.

**Comment by MaxGekk** [comment]
@itholic Please, resolve conflicts.

**Comment by MaxGekk** [comment]
@itholic Could you fix the test failures:
```
[info] - SPARK-34057: UnresolvedTableOrView should retain sql text position *** FAILED *** (139 milliseconds)
[info]   Expected exception org.apache.spark.sql.AnalysisException to be thrown, but org.apache.spark.SparkException was thrown (AnalysisTest.scala:182)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
```

**Comment by MaxGekk** [comment]
@itholic Please, resolve conflicts and fix tests.

**Comment by MaxGekk** [comment]
@itholic Can you fix GAs? Some test failures are related to your changes:
```
2022-10-30 23:01:03.735 - stderr> org.apache.spark.SparkException: spark.sql.catalog.default
2022-10-30 23:01:03.735 - stderr> 	at org.apache.spark.sql.errors.QueryExecutionErrors$.noSuchElementExceptionError(QueryExecutionErrors.scala:2178)
2022-10-30 23:01:03.736 - stderr> 	at org.apache.spark.sql.internal.SQLConf.$anonfun$getConfString$3(SQLConf.scala:4839)
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 1]
```diff
@@ -2171,11 +2168,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
   }
 
   def noSuchElementExceptionError(key: String): Throwable = {
-    new SparkException(
-      errorClass = "_LEGACY_ERROR_TEMP_2221",
-      messageParameters = Map(
-        "key" -> key),
-      cause = null)
+    new NoSuchElementException(key)
```
Comment (itholic) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:2171]: Seems like we cannot just simply treat `key` for `NoSuchElementException` as same as message in error class.

I think maybe we need to new Exception something like `SparkNoSuchElementException`.

Let me just revert for now, and will open a PR for new Exception, and cover the missing cases here.

**Comment by itholic** [comment]
@MaxGekk CI passed finally..😂   Could you take a look when you find some time ?

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Pull request closed without merging**